### PR TITLE
add generateAddress2 for CREATE2

### PR DESCRIPTION
**Pull request by jwasinger**
Precursor to [EIP 1014](https://github.com/ethereum/EIPs/pull/1014) implementation on EthereumJS VM .  Need to add unit tests.

There is currently an open PR [#1375](https://github.com/ethereum/EIPs/pull/1375) on the EIPs repository which represents the more accurate state of the EIP.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -531,6 +531,27 @@ exports.generateAddress = function (from, nonce) {
...
+ * @param {Buffer} inputCode the code of the contract being created
+ * @return {Buffer}
+ */
+exports.generateAddress2 = function (from, salt, inputCode) {
+  from = exports.toBuffer(from)
+  salt = exports.toBufer(salt)
+  inputCode = exports.toBuffer(inputCode)
+
+  let address = exports.keccak256(Buffer.concat([
+    Buffer.from('0xff', 'hex'),
```
Comment (rmeissner) [inline index.js]: shouldn't `from` be part of the hash?

as mentioned in https://github.com/ethereum/EIPs/pull/1014#issuecomment-412106601

Comment (jwasinger) [reply index.js]: Yup.. good call.

**Comment by holgerd77** [comment]
This needs to be updated and the tests fixed.

**Comment by holgerd77** [comment]
Have updated this according to the latest discussions. Existing tests should now run through, this still needs specific test cases for the new function though. Have asked on the EIP update PR for some examples.

**Comment by holgerd77** [comment]
Update: examples on the EIP update PR page (should be merged soon) are now ready, can be added as test cases here.

File: index.js [hunk 2]
```diff
@@ -531,6 +531,28 @@ exports.generateAddress = function (from, nonce) {
...
+/**
+ * Generates an address for a contract created using CREATE2
+ * @param {Buffer} from the address which is creating this new address
+ * @param {Buffer} salt a salt
+ * @param {Buffer} initCode the init code of the contract being created
+ * @return {Buffer}
+ */
+exports.generateAddress2 = function (from, salt, initCode) {
+  from = exports.toBuffer(from)
+  salt = exports.toBuffer(salt)
```
Comment (rmeissner) [inline index.js]: Should the length of the buffers be checked (as they should be 20 and 32, else the hashes would differ, right?)

**Comment by holgerd77** [comment]
Yes, that makes sense

**Comment by holgerd77** [comment]
@rmeissner Ok, updated the example tests file and will merge once CI is through. Won't make it to do a release today, will have to think a bit more thoroughly on how to organize that and what to put inside. Hopefully tomorrow.

**Comment by holgerd77** [comment]
Current idea would be to merge https://github.com/ethereumjs/ethereumjs-util/pull/143 (EIP-155 support), merge this one and then do a ``v5.3.0`` release.

**Pull request merged**